### PR TITLE
fix: prevent duplicate terms when splitting shorter than term_splitter_count

### PR DESCRIPTION
**Pull request by zurchpet**
- Fix split method to avoid duplicates when term length is shorter than term_splitter_count
- Add max_length calculation to limit splitting to actual term length
- Add comprehensive RSpec tests for edge cases:
  - Terms shorter than term_splitter_count
  - Empty terms handling
  - Multiple spaces handling
  - Empty string input

### CODE REVIEW

File: lib/xapian_db/indexer.rb [hunk 1]
```diff
@@ -125,7 +125,8 @@ def get_values_to_index_from(obj)
     def split(terms)
       splitted_terms = []
       terms.split(" ").each do |term|
```
[the remark this answers is not in the document]
Comment (zurchpet) [reply lib/xapian_db/indexer.rb:127]: There is no need to `reject(&:empty?` after a  `.split(" ")`.

**Comment by btihen** [comment]
👍

**Comment by gernotkogler** [comment]
Released 1.3.17
